code/model1.py

```diff
@@ -1,9 +1,76 @@
 from model import Model
+import torch
+import torch.nn as nn
+import torch.autograd as autograd
+import torch.optim as optim
+import utils
+import numpy as np
 
 
-class Model1(Model):
+class Model1(nn.Module, Model):
+
+    def __init__(self, epochs=10):
+        self.epochs = epochs
+        super(Model1, self).__init__()
+        # embedding lookup for 150 users each have 100 dimension representation
+        self.embedding_layer = nn.Embedding(150, 100)
```

Comment:
Let's reduce this to something much smaller like 20 or 50

Reply:
yeah ... it was just an approx number